subprojects/packagefiles/theora/meson.build

```diff
@@ -0,0 +1,224 @@
+
+if not oss_dep.found()
+  message('OSS audio support not found -- not compiling player_example')
+endif
+
+libpng_dep = dependency('libpng', required: false)
+
+if cc.has_header('unistd.h')
+  unistd_dep = declare_dependency()
+else
+  unistd_dep = disabler()
+endif
+
+telemetry = get_option('telemetry')
+
+cairo_dep = dependency('cairo', required: telemetry)
+
+if cairo_dep.found()
+  conf.set('HAVE_CAIRO', true)
+endif
+
+mem_constraint = get_option('mem-constraint')
+
+if mem_constraint.enabled()
+  conf.set('HAVE_MEMORY_CONSTRAINT', language: 'c')
+endif
+
+asm = not get_option('asm').disabled()
+if asm
+  if host_machine.cpu_family() == 'x86_64'
+    conf.set('CPU_x86_64', true)
+    if cc.get_id() != 'msvc'
+      conf.set('OC_X86_ASM', true)
+      conf.set('OC_X86_64_ASM', true)
+    endif
+  elif host_machine.cpu_family() == 'x86'
+    conf.set('OC_X86_ASM', true)
+  endif
+endif
+
+config_h = configure_file(configuration: conf, output: 'config.h')
+
+config_dep = declare_dependency(sources: config_h)
+
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+
+pkg = import('pkgconfig')
+windows = import('windows')
+
+subdir('include')
+
+subdir('lib')
+
+if get_option('examples').enabled()
+  subdir('examples')
+endif
+
+subdir('tests')
+
+doc = get_option('doc')
+
+doc_root_dir = meson.current_source_dir()
+
+doxygen_exe = find_program(
+  'doxygen',
+  required: doc,
+  disabler: true,
+)
+
+subdir('doc')
+
+spec = get_option('spec')
+
+if not spec.disabled()
+  # Note: unlike the original autotools script, here we use
+  # latexmk to do the dependency management for us.
+  # This implicates that we've got to copy spec.{tex,bib}
+  # to the build folder in /doc/spec/ to make all files
+  # live in the same folder.
+  latexmk_exe = find_program('latexmk', required: spec)
+  transfig_exe = find_program('fig2dev', required: spec)
+  # used indirectly by latexmk
+  pdflatex = find_program('pdflatex', required: spec)
+  bibtex = find_program('bibtex', required: spec)
+
+  if (
+    latexmk_exe.found()
+    and pdflatex.found()
+    and bibtex.found()
+    and transfig_exe.found()
+  )
+    check_needed_tex_packages = find_program(
+      'doc/spec/check-needed-tex-packages.py',
+      required: true,
+    )
+    test_tex_packages = find_program(
+      'doc/spec/test-tex-packages.py',
+      required: true,
+    )
+
+    message(
+      'Checking for packages in @0@...'.format(
+        meson.current_source_dir() / 'doc/spec/spec.tex',
+      ),
+    )
+
+    packages = run_command(
+      check_needed_tex_packages,
+      capture: true,
+      check: true,
+    ).stdout().split()
+
+    tex_pkg_ok = true
+
+    foreach pkg : packages
+      message('for Tex package @0@'.format(pkg))
+
+      pkg_test = run_command(
+        test_tex_packages,
+        '--pdflatex',
+        pdflatex,
+        pkg,
+        check: spec.enabled(),
+      )
+      if pkg_test.returncode() != 0
+        tex_pkg_ok = false
+        break
+      endif
```

Comment:
But over here, you've finished doing some manual checks for tex packages, and discovered that the spec cannot be built even though the programs are available. What if spec was enabled, instead of auto? This should probably be:

```meson
if pkg_test.returncode() != 0
  if spec.enabled()
    error('Format Specification is required, but cannot be built due to missing tex packages')
  else
    tex_pkg_ok = false
    break
  endif
endif

Reply:
The `run_command` above was designed to [error out](https://mesonbuild.com/Reference-manual_functions.html#run_command), as I'm using `check: spec.enabled()`.

e.g. with -Dspec=auto 
```
Program doc/spec/check-needed-tex-packages.py found: YES (C:\Program Files\Python310\python.exe E:\wrapdb\subprojects\libtheora-1.1.1\doc/spec/check-needed-tex-packages.py)
Program doc/spec/test-tex-packages.py found: YES (C:\Program Files\Python310\python.exe E:\wrapdb\subprojects\libtheora-1.1.1\doc/spec/test-tex-packages.py)
Message: Checking for packages in E:/wrapdb/subprojects/libtheora-1.1.1/doc/spec/spec.tex...
Message: for Tex package latexsym
Message: *** Format Specification will not built.
```

and with `-Dspec=enabled`
```
Message: Checking for packages in E:/wrapdb/subprojects/libtheora-1.1.1/doc/spec/spec.tex...
Message: for Tex package latexsym

meson.build:197:6: ERROR: Command "C:\Program Files\Python310\python.exe E:\wrapdb\subprojects\libtheora-1.1.1\doc/spec/test-tex-packages.py --pdflatex D:\texlive\2022\bin\win32\pdflatex.EXE 
latexsym" failed with status 1.

A full log can be found at E:\wrapdb\subprojects\libtheora-1.1.1\build\meson-logs\meson-log.txt
```

Your alternative does make the error more obvious, so I'll take it.

Reply:
Right. I missed that indeed. But the error message is superior anyway. :D